### ISSUE TITLE
`C-c C-SPC` (give (fill goal)) fails on a perfectly valid filler with an error message mentioning non-existent variables when `C-u C-u C-c C-SPC` succeeds

### ISSUE THREAD

**Issue by JasonGross** [agda/agda#5307]
```agda
module foo where
record Cat : Set₂ where
  constructor sproc
  field Obj : Set₁
  field Arr : Obj -> Obj -> Set
  field Arr² : ∀ {a b} → Arr a b → Arr a b → Set
  record Iso² {a b} (f g : Arr a b) : Set where
    constructor iso²
    field fwd : Arr² f g
    field bak : Arr² g f
_^op : Cat → Cat
_^op A = record
  { Obj = A.Obj
  ; Arr = λ{ a b → A.Arr b a }
  ; Arr² = A.Arr²
  } where module A = Cat A
op-imap² : ∀ {A} {a₁ a₂ : Cat.Obj A} {p₁ p₂ : Cat.Arr A a₁ a₂} → Cat.Iso² A p₁ p₂ → Cat.Iso² (A ^op) p₁ p₂
op-imap² f = record
  { fwd = f.fwd
  ; bak = f.bak
  } where module f = Cat.Iso² f
module _ {D : Cat} where
  module D = Cat D
  ans'  : ∀ d (m : D.Arr d d) (pf : D.Iso² m m) → Cat.Iso² (D ^op) m m
  ans' d m pf = op-imap² pf
  postulate ans''  : ∀ d (m : D.Arr d d) (pf : D.Iso² m m) → Cat.Iso² (D ^op) m m
  ans   : ∀ d (m : D.Arr d d) (pf : D.Iso² m m) → Cat.Iso² (D ^op) m m
  ans = {!ans'!}
{- C-c C-x C-h C-c C-SPC gives:
/home/jgross/Documents/foo.agda:29,11-15
x != x₁ of type D.Obj
when checking that the expression ans' has type
(d : D.Obj) (m : D.Arr d d) →
D.Iso² {d} {d} m m → Cat.Iso² (D ^op) {d} {d} m m
-}
```
Using `ans''` instead works fine.  Using `C-u C-u C-c C-SPACE` also works fine  Possibly related to this is that, in my original example, `C-c C-x C-h C-c C-t` gets the type of the goal wrong, in the sense that it ignores the implicits that I specify explicitly and instead inserts βιδ-expanded versions of them (the versions that would show up if it were re-inferring the implicits from scratch).  However, I cannot reproduce this in my minimized example, and my original example was somewhat more complicated, so I'm a bit disinclined to try to minimize this possibly related bug.

What's going on here?

```
$ agda --version
Agda version 2.6.1.2
```

**Comment by JasonGross** [comment]
This seems to be an issue with Agda performing eta-contraction without checking that the constructor arguments which are record parameters line up, given that this is a smaller example:
```agda
module foo where
record Foo : Set₂ where
  field foo : Set₁
record Bar (f : Foo) : Set₂ where
  field bar : Set₁
resize : ∀ f g → Bar f → Bar g
resize f g b = record { bar = Bar.bar b }

resize' : ∀ f g → Bar f → Bar g
resize' f g b = {!resize f g b!}
```

**Comment by jespercockx** [comment]
Both your examples seem to be working on the current master. #5308 is still open though, so I'm not sure what changed to fix this.

**Comment by jespercockx** [comment]
Closing this due to a lack of response.

**Comment by andreasabel** [comment]
I could reproduce the OP with 2.6.0 and 2.6.1, but it is fixed in 2.6.2.
